noir_stdlib/src/hash.nr

```diff
@@ -25,22 +26,17 @@ pub fn blake3<N>(input: [u8; N]) -> [u8; 32]
 {}
 
 // docs:start:pedersen_commitment
-struct PedersenPoint {
-   x : Field,
-   y : Field,
-}
-
-pub fn pedersen_commitment<N>(input: [Field; N]) -> PedersenPoint {
+pub fn pedersen_commitment<N>(input: [Field; N]) -> EmbeddedCurvePoint {
     // docs:end:pedersen_commitment
     pedersen_commitment_with_separator(input, 0)
 }
 
 #[foreign(pedersen_commitment)]
 pub fn __pedersen_commitment_with_separator<N>(input: [Field; N], separator: u32) -> [Field; 2] {}
 
-pub fn pedersen_commitment_with_separator<N>(input: [Field; N], separator: u32) -> PedersenPoint {
+pub fn pedersen_commitment_with_separator<N>(input: [Field; N], separator: u32) -> EmbeddedCurvePoint {
     let values = __pedersen_commitment_with_separator(input, separator);
-    PedersenPoint { x: values[0], y: values[1] }
+    EmbeddedCurvePoint { x: values[0], y: values[1], is_infinite: false }
```

Comment:
We can have a pedersen commitment which returns the point at infinity (e.g. a commitment to all zero values) so this shows that we're not fully representing the range of potential outputs.

```rust
 let pedersen_commitment_to_zero = std::hash::pedersen_commitment([0]);
println(pedersen_commitment_to_zero);
```
executing this code gets the panic
```
% nargo execute   ~/noir/test_programs/execution_success/pedersen_check gd/issue_4931 + tom-box
The application panicked (crashed).
Message:  should not commit to point at infinity
Location: acvm-repo/bn254_blackbox_solver/src/lib.rs:48
```